### PR TITLE
GBufferGroup hsTArray -> std::vector

### PR DESCRIPTION
**Pull request by Hoikas**
Does what it says. `std::vector` is just better than `hsTArray`.

### CODE REVIEW

**Comment by dpogue** [comment]
:+1:

File: Sources/Plasma/PubUtilLib/plPipeline/plGBufferGroup.cpp [hunk 1]
```diff
@@ -439,24 +417,24 @@ void    plGBufferGroup::Read( hsStream *s )
     for( i = 0; i < count; i++ )
     {
         temp = s->ReadLE32();
-        fIdxBuffCounts.Append( temp );
-        fIdxBuffStarts.Append(0);
-        fIdxBuffEnds.Append(-1);
+        fIdxBuffCounts.push_back(temp);
+        fIdxBuffStarts.push_back(0);
+        fIdxBuffEnds.push_back(-1);
```
Comment (zrax) [inline Sources/Plasma/PubUtilLib/plPipeline/plGBufferGroup.cpp]: Whenever you see `push_back` in a known-count loop, it's a good idea to `reserve` the space in the vector first...

File: Sources/Plasma/PubUtilLib/plPipeline/plGBufferGroup.h [hunk 2]
```diff
@@ -121,24 +121,24 @@ class plGBufferGroup
...
-        hsTArray<plGBufferColor *>  fColorBuffStorage;
+        std::vector<uint32_t> fVertBuffStarts;
+        std::vector<int32_t>  fVertBuffEnds;
+        std::vector<uint32_t> fIdxBuffStarts;
+        std::vector<int32_t>  fIdxBuffEnds;
 
-        hsTArray<hsTArray<plGBufferCell> *> fCells;
+        std::vector<plGBufferColor*>  fColorBuffStorage;
+
+        std::vector<std::vector<plGBufferCell>*> fCells;
```
Comment (zrax) [inline Sources/Plasma/PubUtilLib/plPipeline/plGBufferGroup.h]: Do these really need to be pointers to vectors?  I'm thinking they don't...

**Comment by Hoikas** [comment]
Updated to reserve space in vectors and get rid of `std::vector<plGBufferCell>*`

**Pull request merged**